### PR TITLE
LPD-15620

### PR DESCRIPTION
**Pull request by rijema**
https://liferay.atlassian.net/browse/LPD-15620

The customer is informing that a data provider object is retaining a huge value on their memory. The intention is to execute the data process faster  (defining a limit to its memory) and treat (on a system configuration level) the GC properties to deal better with old objects. I had good results by changing the setent.sh file on the web server folder and starting the portal with this configuration:

`CATALINA_OPTS="$CATALINA_OPTS -Dfile.encoding=UTF-8 -Djava.locale.providers=JRE,COMPAT,CLDR -Djava.net.preferIPv4Stack=true -Duser.timezone=GMT -XX:NewSize=700m -XX:MaxNewSize=700m -Xms2048m -Xmx2048m -XX:MaxPermSize=128m -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:SurvivorRatio=20 -XX:ParallelGCThreads=8"`

Analysis: https://github.com/Netflix/Hystrix/wiki/Configuration 

Let me know the necessary changes. If passed review, please let me know which test to implement on this case. 

Thanks =)

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java [hunk 1]
```diff
@@ -43,6 +43,9 @@ public DDMDataProviderInvokeCommand(
 				HystrixCommandProperties.Setter().
 					withExecutionTimeoutInMilliseconds(
 						_getTimeout(ddmRESTDataProviderSettings))
+			).andCommandPropertiesDefaults(
+			   HystrixCommandProperties.Setter()
+					.withExecutionIsolationSemaphoreMaxConcurrentRequests(20)
 			));
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java:49]: @rijema please reuse the existing `andCommandPropertiesDefaults`:
```
).andCommandPropertiesDefaults(
	HystrixCommandProperties.Setter().
		withExecutionIsolationSemaphoreMaxConcurrentRequests(20)
		.withExecutionTimeoutInMilliseconds(
			_getTimeout(ddmRESTDataProviderSettings))
));
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java [hunk 2]
```diff
@@ -43,6 +43,9 @@ public DDMDataProviderInvokeCommand(
 				HystrixCommandProperties.Setter().
 					withExecutionTimeoutInMilliseconds(
 						_getTimeout(ddmRESTDataProviderSettings))
+			).andCommandPropertiesDefaults(
+			   HystrixCommandProperties.Setter()
+					.withExecutionIsolationSemaphoreMaxConcurrentRequests(20)
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java:48]: @rijema How did you choose this number 20?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/build.gradle [hunk 3]
```diff
@@ -1,7 +1,7 @@
 dependencies {
 	compileInclude group: "com.jayway.jsonpath", name: "json-path", version: "2.2.0"
 	compileInclude group: "com.netflix.archaius", name: "archaius-core", version: "0.4.1"
-	compileInclude group: "com.netflix.hystrix", name: "hystrix-core", version: "1.5.11"
+	compileInclude group: "com.netflix.hystrix", name: "hystrix-core", version: "1.5.18"
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/build.gradle:4]: @rijema Is there any reason why you decided to update the version?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java (hunk 1 above)
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java:49]: Ok. I will chain it.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java (hunk 2 above)
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java:48]: According to the documentation, the Default Value is 10. https://github.com/Netflix/Hystrix/wiki/Configuration#execution.isolation.semaphore.maxConcurrentRequests

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/build.gradle (hunk 3 above)
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/build.gradle:4]: This is the latest version of this lib, released 6 years ago. The highest usage is concentrated on this version.  I noticed that until the .13 version some adjustments were done on the code, and I considered to ensure the quality of the lib on our portal by also upgrading and add these modifications. 

https://github.com/Netflix/Hystrix/tags

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokeCommand.java:48]: @rijema I saw it, but why did you change it to 20? Did you perform some test to define this number?

**Comment by linolaoj** [comment]
Hi @rijema , as @carolmariaabb mentioned about the `ExecutionIsolationSemaphoreMaxConcurrentRequests` it is important that the information of the values with and without the changes is added here so it is clear the difference in memory size. Also, in the LPD-15620, it is mentioned in the actual and expected results "_high levels_", it is better if we had some actual numbers to compare results.

**Comment by linolaoj** [comment]
@rijema , do we need the configuration of GC that you mention to see the memory results in Hystrix? Or is it just to enhance the result of the tuning?

**Comment by rijema** [comment]
Hi @linolaoj @carolmariaabb, the new tests were being inconclusive, that's the reason for not updating y'all on this PR. Can we sit and talk for a few minutes today?

**Comment by linolaoj** [comment]
Hi @rijema we can talk about it. I believe we can close this PR for now.

**Pull request closed without merging**